pkg/alertmanager/operator.go

```diff
@@ -1040,6 +1044,21 @@ func (c *Operator) selectAlertmanagerConfigs(ctx context.Context, am *monitoring
 	return res, nil
 }
 
+func checkAlertmanagerRootConfig(ctx context.Context, conf *alertmanagerConfig) error {
+	if conf == nil || conf.Route == nil {
+		return nil
+	}
+
+	if len(conf.Route.Matchers) > 0 {
+		return errors.Errorf("'matchers' not permitted on root route: %v", conf.Route.Matchers)
+	}
+
+	if len(conf.Route.MuteTimeIntervals) > 0 {
+		return errors.Errorf("'mute_time_intervals' not permitted on root route: %v", conf.Route.MuteTimeIntervals)
+	}
+	return nil
+}
+
 // checkAlertmanagerConfig verifies that an AlertmanagerConfig object is valid
 // and has no missing references to other objects.
 func checkAlertmanagerConfig(ctx context.Context, amc *monitoringv1alpha1.AlertmanagerConfig, amVersion semver.Version, store *assets.Store) error {
```

Comment:
side-note: I find that the function name is confusing, my first instinct it to assume that it checks the Alertmanager global configuration. What about checkAlertmanagerConfigResource instead?

Reply:
yes, when I added the other func I thought the same, I expected this to take care of it at first glance. will update